### ISSUE TITLE
There was some trouble enabling the  Font Ligatures

### ISSUE THREAD

**Issue by netmou** [microsoft/vscode#81201]
php code :
```php
if (!preg_match('/[!\\(\\)><=\\s]+/s', trim($val))) {
  trigger_error("Expression: ‘{$val}’ may be invalid！", E_USER_ERROR);
}
```
in vscode looks like  :
![image](https://user-images.githubusercontent.com/7068058/65291700-1de31500-db87-11e9-9c84-a2cc904cd031.png)

the '<=' in string '/[!\\(\\)><=\\s]+/s' just means a list,not 'not equal' 
I suggest that when the boundary of a Ligatures character is still a not word character or not an operand, it should be closed, so for example, '1=a ' can be opened , but ' !>=\\' should not be opened

In other words, the casual use of font Ligatures without semantics leads to ambiguity. The original is more beautiful than the false

**Comment by kieferrm** [comment]
@netmou I'm having trouble understanding your point. Can you please explain in more detail.

**Comment by netmou** [comment]
the '<=' in string '/[!\(\)><=\s]+/s' just means a list,not 'not equal'

**Comment by netmou** [comment]
sorry for my english is not good

**Comment by netmou** [comment]
Some ligatures still enabled in regex groups

**Comment by chrmarti** [comment]
My understanding is that we do not control ligatures and this is an upstream issue. @alexandrudima ?

**Comment by alexandrudima** [comment]
That is correct, font rendering is done at a lower level than the syntax... Whenever `<` and `=` appear one after another, they get merged via ligatures in a single symbol.
